### DISCUSSION TITLE
Any reason for not being able to prevent auto initialization of HSTooltip?

### DISCUSSION THREAD

**Discussion by sebz** [htmlstreamofficial/preline#402] [Q&A] [answered]
Hey there! 

I'm integrating Preline in an Elixir Phoenix app and while trying to avoid tooltips to get created multiple times I tried to use the `--prevent-on-load-init` option but it appears HSTooltip does not support it... why is that?

Thanks for your help 🙏

**Comment by langscot** [comment, marked as answer]
Seems like the HSTooltip's `autoInit` method does not exclude elements with `--prevent-on-load-init`class.

```ts
	static autoInit() {
		if (!window.$hsTooltipCollection) window.$hsTooltipCollection = [];

		document.querySelectorAll('.hs-tooltip').forEach((el: HTMLElement) => {
			if (
				!window.$hsTooltipCollection.find(
					(elC) => (elC?.element?.el as HTMLElement) === el,
				)
			)
				new HSTooltip(el);
		});
	}
```

This should be an open issue, I can't think why it wouldn't respect `--prevent-on-load-init` since as far as I'm aware the other plugins do. 

As a workaround, you could not `autoInit` the `tooltip` collection and do this yourself somewhere in your code:

```ts
document.querySelectorAll('.hs-tooltip:not(.--prevent-on-load-init)').forEach((el: HTMLElement) => {
	if (
		!window.$hsTooltipCollection.find(
			(elC) => (elC?.element?.el as HTMLElement) === el,
		)
	)
		new HSTooltip(el);
});
```

**Reply by sebz** [reply]
Thanks @langscot and sorry for the very delayed answer... I somehow missed the notification 😑
I indeed implemented a similar workaround 😅 Thanks for taking the time🙏 

I'll open an issue since I see than in `2.5.1` it's still not supported.
